shell/egl_window.cc

```diff
@@ -62,39 +56,29 @@ EglWindow::EglWindow(size_t index,
       xdg_wm_base_get_xdg_surface(m_display->GetXdgWmBase(), m_surface);
   xdg_surface_add_listener(m_xdg_surface, &xdg_surface_listener, this);
   m_xdg_toplevel = xdg_surface_get_toplevel(m_xdg_surface);
+  xdg_toplevel_add_listener(m_xdg_toplevel, &xdg_toplevel_listener, this);
   xdg_toplevel_set_title(m_xdg_toplevel, m_app_id.c_str());
 
   m_display->WaitForConfig();
+  wl_surface_commit(m_surface);
 
-  m_width = (m_sprawl && m_display->GetModeWidth() > 0)
-                ? m_display->GetModeWidth()
-                : width;
-  m_height = (m_sprawl && m_display->GetModeHeight() > 0)
-                 ? m_display->GetModeHeight()
-                 : height;
-
-  m_egl_window[index] = wl_egl_window_create(m_surface, m_width, m_height);
-
-  FML_DLOG(INFO) << "create egl_window: " << m_width << "x" << m_height;
-
-  m_egl_surface[index] = create_egl_surface(this, m_egl_window[index], nullptr);
-
-  set_fullscreen(m_fullscreen);
-
-  /* Initialise damage to full surface, so the padding gets painted */
-  wl_surface_damage(m_surface, 0, 0, m_width, m_height);
+  m_egl_window[m_index] =
+      wl_egl_window_create(m_surface, m_geometry.width, m_geometry.height);
```

Comment:
The m_geometry.width and m_geometry.height are passed from App() right? 

In practice, are these the updated values from the resize  in top level configure or the initial ones?

Have you tried these in system with software rendering?

Reply:
The initial values passed from App() get updated via handle_toplevel_configure().

I have not yet tried this PR with a software render (Qemu).  I need to validate it on my CI platforms (meta-flutter).

I am planning at some point to add optional support for a decoration library for a more friendly desktop support.  So resizing etc will be added later.  The primary use case is a minimal target app, that happens to work on desktop.